### ISSUE TITLE
[Bug] Strings cause memory free errors & segfaults.

### ISSUE THREAD

**Issue by SirJosh3917** [theduke/quickjs-rs#105]
Minimal repro:

```rust
fn main() {
    let ctx = quick_js::Context::new().unwrap();
    ctx.set_global("a", "a").unwrap();
    ctx.eval("a").unwrap();
}
```

```
double free or corruption (!prev)
Aborted
```

```
Segmentation fault
```

Performing a computation causes a segfault:

```rust
fn main() {
    let ctx = quick_js::Context::new().unwrap();
    ctx.set_global("a", "a").unwrap();
    ctx.eval("a + 1").unwrap();
}
```

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Exception(String("InternalError: string too long"))', src/main.rs:4:23
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Segmentation fault
```

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Exception(String("InternalError: string too long"))', src/main.rs:4:23
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
free(): invalid pointer
Aborted
```

Tested it on two `x86_64-pc-linux-gnu` machines. GCC `9.3.0` on both, and rustc `1.50.0-nightly`/`1.47.0`. Not sure what other info is needed to reproduce.

**Comment by theduke** [comment]
Can't reproduce on Linux.

Could you run valgrind? 

`valgrind --tool=memcheck /target/release/NAME-OF-BINARY`?

**Comment by theduke** [comment]
Never mind, found the issue.

**Comment by theduke** [comment]
Released as `0.4.1`.
